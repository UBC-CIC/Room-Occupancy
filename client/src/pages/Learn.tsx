import * as React from "react";
import { Header, Grid, Image } from "semantic-ui-react";
import "../App.css";
import learn from "../assets/images/learn_1.png";

export interface ILearnProps {}

export function Learn(props: ILearnProps) {
  return (
    <div className="container">
      <Header color="blue" as="h1" textAlign="center">
        <Header.Content>Learn</Header.Content>
      </Header>
      <br />
      <Grid>
        <Grid.Row columns={2}>
          <Grid.Column>
            The UBC Cloud Innovation Center seeks a cutting-edge solution to
            address the challenge of real-time people counting and occupancy
            monitoring in spaces, such as rooms or traffic areas. This solution
            should utilize IoT devices, an open-source image processor, and
            cloud services to accurately count and track the number of people in
            these areas. The data gathered should be presented via a
            user-friendly interface that efficiently conveys information.
          </Grid.Column>
          <Grid.Column>
            <Image size="big" src={learn} />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </div>
  );
}
