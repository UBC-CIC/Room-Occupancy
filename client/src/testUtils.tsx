import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import {
  MemoryRouter,
  MemoryRouterProps,
  BrowserRouter,
} from "react-router-dom";

// Implement with creation of providers
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <MemoryRouter>{children}</MemoryRouter>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

// Usage example:
// test("matches snapshot", () => {
//   // Replace 'Navbar' with your desired component and provide the props
//   testSnapshot(Navbar, { /* props for Navbar */ });
// });
const testComponentSnapshot = <P extends {}>(
  Component: React.ComponentType<P>,
  props: P,
  routerProps: MemoryRouterProps = {}
) => {
  const { asFragment } = render(
    <MemoryRouter {...routerProps}>
      <Component {...props} />
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
};

export * from "@testing-library/react";
export { customRender, testComponentSnapshot };
