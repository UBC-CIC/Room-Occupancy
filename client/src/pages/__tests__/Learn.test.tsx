import * as React from "react";
import { render, act, screen } from "@testing-library/react";
import { customRender } from "../../testUtils";
import { Learn } from "../Learn";
import "@testing-library/jest-dom";
import { BrowserRouter as Router } from "react-router-dom";

describe("<Learn />", () => {
  it("matches snapshot", () => {
    const { asFragment } = customRender(<Learn />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders learn paragraph", () => {
    act(() => {
      customRender(<Learn />);
    });
    const learnParagraph = screen.getByText(
      "The UBC Cloud Innovation Center seeks a cutting-edge solution to address the challenge of real-time people counting and occupancy monitoring in spaces, such as rooms or traffic areas. This solution should utilize IoT devices, an open-source image processor, and cloud services to accurately count and track the number of people in these areas. The data gathered should be presented via a user-friendly interface that efficiently conveys information."
    );
    expect(learnParagraph).toBeInTheDocument();
  });
});
