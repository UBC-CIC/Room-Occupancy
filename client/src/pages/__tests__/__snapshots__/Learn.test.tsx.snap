// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Learn /> matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbarContainer"
  >
    <div
      class="ui mini basic attached segment"
      fluid="true"
    >
      <div
        class="ui grid"
      >
        <div
          class="six wide column"
        >
          <img
            alt="logo"
            src="test-file-stub"
            width="300rem"
          />
        </div>
        <div
          class="center aligned five wide column"
        >
          <div
            basic="true"
            centered="true"
            class="ui secondary menu"
          >
            <a
              href="/"
            >
              <div
                class="item"
                role="menuitem"
              >
                Home
              </div>
            </a>
            <a
              href="/learn"
            >
              <div
                class="item"
                role="menuitem"
              >
                Learn
              </div>
            </a>
            <a
              href="/about"
            >
              <div
                class="item"
                role="menuitem"
              >
                About
              </div>
            </a>
            <a
              href="/analytics"
            >
              <div
                class="item"
                role="menuitem"
              >
                Analytics
              </div>
            </a>
          </div>
        </div>
        <div
          class="right aligned five wide column"
        >
          <div
            class="ui buttons"
          >
            <a
              href="/login"
            >
              <button
                class="ui button"
              >
                Login
              </button>
            </a>
            <div
              class="or"
              data-text="or"
            />
            <a
              href="/signup"
            >
              <button
                class="ui blue button"
              >
                SignUp
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="container"
  >
    <h1
      class="ui blue center aligned header"
    >
      <div
        class="content"
      >
        Learn
      </div>
    </h1>
    <br />
    <div
      class="ui grid"
    >
      <div
        class="two column row"
      >
        <div
          class="column"
        >
          The UBC Cloud Innovation Center seeks a cutting-edge solution to address the challenge of real-time people counting and occupancy monitoring in spaces, such as rooms or traffic areas. This solution should utilize IoT devices, an open-source image processor, and cloud services to accurately count and track the number of people in these areas. The data gathered should be presented via a user-friendly interface that efficiently conveys information.
        </div>
        <div
          class="column"
        >
          <img
            class="ui big image"
            src="test-file-stub"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
